backend/x/gorgonnx/unsqueeze.go

```diff
@@ -0,0 +1,56 @@
+package gorgonnx
+
+import (
+	"github.com/owulveryck/onnx-go"
+	"gorgonia.org/gorgonia"
+)
+
+// SPEC: https://github.com/onnx/onnx/blob/master/docs/Operators.md#Unsqueeze
+
+type unsqueeze struct {
+	Axes []int64
+}
+
+func init() {
+	register("Unsqueeze", newUnsqueeze)
+}
+func newUnsqueeze() operator {
+	return &unsqueeze{}
+}
+
+func (a *unsqueeze) apply(g *Graph, ns ...*Node) error {
+	n := ns[0]
+	children := getOrderedChildren(g.g, n)
+	err := checkCondition(children, 1)
+	if err != nil {
+		return err
+	}
+
+	tensor := children[0].gorgoniaNode
+	dims := make([]int, len(a.Axes)+tensor.Dims())
+	for k := range dims {
+		dims[k] = -1
+	}
+	for _, v := range a.Axes {
+		dims[v] = 1
+	}
+	var index int
+	for k, v := range dims {
+		if v == -1 {
+			index = k
+			break
+		}
+	}
+	for i := 0; i < tensor.Dims(); i++ {
+		dims[i+index] = tensor.Shape()[i]
+	}
+	n.gorgoniaNode, err = gorgonia.Reshape(tensor, dims)
+
+	return err
+
+}
+
+func (a *unsqueeze) init(o onnx.Operation) error {
+	a.Axes, _ = o.Attributes["Axes"].([]int64)
```

Comment:
This is the failing test:
```text
✗  go test -run=ONNX/Unsqueeze
--- FAIL: TestONNX (0.01s)
    --- FAIL: TestONNX/TestUnsqueeze (0.00s)
        test_structure.go:136: the two tensors doesn't have the same dimension, expected (1, 3, 4, 5), got (3, 4, 5)
FAIL
exit status 1
FAIL    github.com/owulveryck/onnx-go/backend/x/gorgonnx        0.018s
```

It is because the `Axes` argument specified by onnx is actually [`axes`](https://github.com/onnx/onnx/blob/master/docs/Operators.md#attributes-78) (note the lowercase a); As a consequence, in this code, `a.Axes` is always a nil array.

I propose to add a test such as:
```go

func (a *unsqueeze) init(o onnx.Operation) error {
        var ok bool
        a.Axes, ok = o.Attributes["axes"].([]int64)
        if !ok {
                return errors.New("unsqueeze: axes in not an []int64")
        }
        return nil
}
```

WDYT?

Reply:
Right. Sorry, I did not check my code.